File: integration/snippets/python/anthropic_env.py
#!/usr/bin/env python3
import os
import sys

os.environ.setdefault("ANTHROPIC_BASE_URL", os.environ.get("PROXY_BASE_URL", ""))
os.environ.setdefault("ANTHROPIC_API_KEY", os.environ.get("PROXY_API_KEY", ""))

if not os.environ["ANTHROPIC_BASE_URL"] or not os.environ["ANTHROPIC_API_KEY"]:
    print("PROXY_BASE_URL and PROXY_API_KEY required", file=sys.stderr)
    sys.exit(1)

from anthropic import Anthropic

client = Anthropic()
msg = client.messages.create(
    model="claude-sonnet-4-5",
    max_tokens=512,
    messages=[{"role": "user", "content": "Hello from the proxy!"}],
)
block = msg.content[0]
print(block.text if block.type == "text" else block)

File: integration/snippets/node/anthropic-env.mjs
process.env.ANTHROPIC_BASE_URL = process.env.PROXY_BASE_URL;
process.env.ANTHROPIC_API_KEY = process.env.PROXY_API_KEY;

if (!process.env.ANTHROPIC_BASE_URL || !process.env.ANTHROPIC_API_KEY) {
  console.error("PROXY_BASE_URL and PROXY_API_KEY required");
  process.exit(1);
}

const { default: Anthropic } = await import("@anthropic-ai/sdk");
const client = new Anthropic();
const msg = await client.messages.create({
  model: "claude-sonnet-4-5",
  max_tokens: 512,
  messages: [{ role: "user", content: "Hello from the proxy!" }],
});
const block = msg.content[0];
console.log(block?.type === "text" ? block.text : JSON.stringify(msg.content));
